Add links and fix some typos and grammar

janriemer opened the pull request:
This adds links to important topics like GitHub issues or Pull requests.
Furthermore, it makes some small formal fixes in the grammar.
Also, it fixes the section on testing for the rust-wasi subsystem.

***Note**: I don't know why the last line is a change. I've edited it in the GitHub UI, so maybe that is why? Maybe a missing new line at the end? I hope this is not a problem. Please let me know.*

Thank you for bringing this project to life! :heart:

janriemer committed:
Add links and fix some typos and grammar

This adds links to important topics like GitHub issues or Pull requests.
Furthermore, it makes some small formal fixes in the grammar.

happybeing commented on CONTRIBUTING.md:
```diff
@@ -83,9 +83,9 @@ Before submitting a pull request affecting the wasi-rust subsystem:
 - if you've modified internal APIs see above re wasi-js subsystem APIs
 - clear up any issues from `cargo clippy -- -D warnings`
 - format using `cargo fmt -- --check`
-- run `yarn test` and make sure all tests pass
```
Although this line looks wrong it should remain `yarn test` for now I think. Later, if we start adding tests in Rust, we can work out how to run them in the existing test suit is `yarn test` or separately as you have here.

happybeing commented:
Thanks very much Jan this is great.

janriemer committed:
`cargo test` -> `yarn test` for rust-wasi

Even though it is Rust, we initially invoke with `yarn`.
Maybe later, we will invoke `cargo test` within `yarn test`.

janriemer commented:
I've upated the section you have mentioned. :wink:
